odoo/orm/models.py

```diff
@@ -2565,55 +2575,56 @@ def _read_group_format_result_properties(self, rows_dict, group):
                 if not row[fullname]:
                     # can not only do ('many2one', '=', False) because we might have
                     # record in database that does not exist anymore
-                    additional_domain = expression.OR([
-                        [(fullname, '=', False)],
-                        [(fullname, 'not in', all_groups)],
-                    ])
+                    additional_domain = [
+                        '|',
+                        (fullname, '=', False),
+                        (fullname, 'not in', all_groups),
+                    ]
                 else:
                     additional_domain = [(fullname, '=', row[fullname])]
                     record = self.env[comodel].browse(row[fullname]).with_prefetch(prefetch_ids)
                     row[fullname] = (row[fullname], record.display_name)
 
-                row['__domain'] = expression.AND([row['__domain'], additional_domain])
+                row['__domain'].extend(additional_domain)
```

Comment:
Wouldn't it be clearer to use a `Domain` for `row['__domain']`, and convert it to a list at the very end?